File: lua/tetris/draw.lua
local m = {}
local g = love.graphics

local SQUARE_SIZE = 25

m.init = function(tetris)
	love.window.setMode(#tetris.field[1] * SQUARE_SIZE, #tetris.field * SQUARE_SIZE)
end

m.figure = function(fig, x, y)
	g.setColor(2, 192, 2)
	for i = 1, #fig do
		for j = 1, #fig[i] do
			if fig[i][j] == 1 then
				g.rectangle("fill",
					(x + j - 1) * SQUARE_SIZE + 1,
					(y + i - 1) * SQUARE_SIZE + 1,
					SQUARE_SIZE - 2, SQUARE_SIZE - 2)
			end
		end
	end
end

m.field = function(field)
	m.figure(field, 0, 0) 
end

m.background = function(field)
	g.setColor(1, 32, 1)
	g.rectangle("fill", 0, 0, #field[1] * SQUARE_SIZE, #field * SQUARE_SIZE)
end

m.tetris = function(tetris)
	local f
	m.background(tetris.field)
	m.field(tetris.field)
	f = tetris.figure
	m.figure(f.f[f.rot].figure, f.x, f.y)
end

return m

File: html5/tetris/draw.ts
module draw {

	const SQUARE_SIZE = 25

	var canvas: HTMLCanvasElement
	var g: CanvasRenderingContext2D

	export function init(tetris: model.Tetris) {
		canvas = <HTMLCanvasElement>document.getElementById('canvas')
		g = <CanvasRenderingContext2D>canvas.getContext('2d')
	}

	export function figure(fig: number[][], x: number, y: number) {
		g.fillStyle = "#02C002"
		for (var i = 0; i < fig.length; ++i) {
			for (var j = 0; j < fig[i].length; ++j) {
				if (fig[i][j] != 0) {
					g.fillRect(
						(x + j) * SQUARE_SIZE + 1,
						(y + i) * SQUARE_SIZE + 1,
						SQUARE_SIZE - 2, SQUARE_SIZE - 2)
				}
			}
		}
	}

	export function field(field: number[][]) {
		figure(field, 0, 0)
	}

	export function background(field: number[][]) {
		g.fillStyle = "#012001"
		g.fillRect(0, 0, field[0].length * SQUARE_SIZE, field.length * SQUARE_SIZE)
	}

	export function tetris(tetris: model.Tetris) {
		var f: model.Figure

		background(tetris.field)
		field(tetris.field)
		f = tetris.figure
		figure(f.f[f.rot].figure, f.x, f.y)
	}
}
